NorthstarDedicatedTest/urihandler.cpp

```diff
@@ -0,0 +1,417 @@
+
+SQRESULT SQ_AcceptInvite(void* sqvm)
+{
+	ClientSq_pushbool(sqvm, HandleAcceptedInvite());
+	hasStoredURI = false;
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_DeclineInvite(void* sqvm)
+{
+	hasStoredURI = false;
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_SetOnMainMenu(void* sqvm)
+{
+	isOnMainMenu = ClientSq_getbool(sqvm, 1);
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_RemoveStoredURI(void* sqvm)
+{
+	hasStoredURI = false;
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_GetInviteServerName(void* sqvm)
+{
+	ClientSq_pushstring(sqvm, g_MasterServerManager->s_requestedServerInfo.name.c_str(), -1);
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_GetInviteShouldShowPasswordDialog(void* sqvm)
+{
+	ClientSq_pushbool(sqvm, (g_MasterServerManager->s_requestedServerInfo.hasPassword && storedPassword.length() == 0));
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_TryJoinInvite(void* sqvm)
+{
+	ClientSq_pushbool(sqvm, HandleAcceptedInvite());
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_ParseInvite(void* sqvm)
+{
+	std::string invite = ClientSq_getstring(sqvm, 1);
+	bool result = parseURI(invite);
+	ClientSq_pushbool(sqvm, result);
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_RequestInviteServerInfo(void* sqvm)
+{
+	g_MasterServerManager->GetServerInfo((char*)storedServerId.c_str(), false);
+	if (!g_MasterServerManager->s_requestedServerInfo.success)
+		lastError = "Failed to get server info";
+	ClientSq_pushbool(sqvm, g_MasterServerManager->s_requestedServerInfo.success);
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_JoinInviteWithPassword(void* sqvm)
+{
+	storedPassword = ClientSq_getstring(sqvm, 1);
+	ClientSq_pushbool(sqvm, HandleAcceptedInvite());
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_GenerateServerInvite(void* sqvm)
+{
+	bool link = ClientSq_getbool(sqvm, 1);
+	std::string id = serverInfo.id;
+	std::string password = serverInfo.password;
+	spdlog::info(password);
+	password = base64_encode(password);
+	spdlog::info(password);
+	std::string invite = "";
+	if (link)
+		invite = Cvar_ns_masterserver_hostname->GetString() + std::string("/invite/server/") + id;
+	else
+	{
+		invite = "northstar://server@" + id;
+	}
+	if (password.length() != 0)
+	{
+		invite += ":" + password;
+	}
+	spdlog::info("GENERATED INVITE: {}, {}", id, password);
+	const size_t inviteLen = strlen(invite.c_str()) + 1;
+	HGLOBAL hMem = GlobalAlloc(GMEM_MOVEABLE, inviteLen);
+	if (hMem != NULL)
+	{
+		LPVOID locked = GlobalLock(hMem);
+		if (locked != NULL)
+		{
+			memcpy(locked, invite.c_str(), inviteLen);
+			GlobalUnlock(hMem);
+			OpenClipboard(0);
+			EmptyClipboard();
+			SetClipboardData(CF_TEXT, hMem);
+			CloseClipboard();
+			ClientSq_pushinteger(sqvm, 1);
+			return SQRESULT_NOTNULL;
+		}
+	}
+	ClientSq_pushinteger(sqvm, 0);
+	return SQRESULT_NOTNULL;
+}
+
+SQRESULT SQ_GetInviteError(void* sqvm)
+{
+	ClientSq_pushstring(sqvm, lastError.c_str(), -1);
+	return SQRESULT_NOTNULL;
+}
+
+void StartUriHandler()
+{
+	spdlog::info("Creating an instance of a named pipe...\n");
```

Comment:
There's quite a few calls to `spdlog::info` with an unnecessary newline in this PR. Perhaps it was originally a `printf` and the newline was missed when replacing it with spdlog?